File: project/jeonga-onboarding/src/main/java/onboarding/survey/dto/UpdateSurveyRequest.java
package onboarding.survey.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public record UpdateSurveyRequest(

        @NotBlank(message = "제목은 필수입니다.")
        String title,

        String description,

        @NotNull(message = "항목 리스트는 필수입니다.")
        @Size(min = 1, max = 10, message = "항목은 1개 이상, 10개 이하로 입력해야 합니다.")
        @Valid
        List<UpdateSurveyItemDto> items
) {
}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/dto/request/UpdateSurveyRequest.kt
package kr.innercircle.onboarding.survey.dto.request

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

/**
 * packageName : kr.innercircle.onboarding.survey.dto.request
 * fileName    : UpdateSurveyRequest
 * author      : ckr
 * date        : 25. 6. 21.
 * description :
 */
data class UpdateSurveyRequest(
    @field:NotBlank(message = "설문조사 이름은 필수값 입니다.")
    val name: String,

    val description: String?,

    @field:Valid
    @field:NotNull
    @field:Size(min = 1, max = 10, message = "설문조사 항목은 최소 1개, 최대 10개까지 가능합니다.")
    val surveyItems: List<UpdateSurveyItemRequest>
)
